hazelcast/src/main/java/com/hazelcast/jet/impl/execution/init/ExecutionPlanBuilder.java

```diff
@@ -88,10 +88,25 @@ public static CompletableFuture<Map<MemberInfo, ExecutionPlan>> createExecutionP
         final int defaultParallelism = nodeEngine.getConfig().getJetConfig().getCooperativeThreadCount();
         final EdgeConfig defaultEdgeConfig = nodeEngine.getConfig().getJetConfig().getDefaultEdgeConfig();
         Set<Integer> requiredPartitions = jobConfig.getArgument(KEY_REQUIRED_PARTITIONS);
+
         final Map<MemberInfo, ExecutionPlan> plans = new HashMap<>();
         int memberIndex = 0;
 
         final Map<MemberInfo, int[]> partitionsByMember = getPartitionAssignment(nodeEngine, memberInfos, requiredPartitions);
+
+        // TODO: ensure that local member (coordinator) belongs to set on which the query is executed.
+        //  This is necessary for RootResultConsumerSink and other forceTotalParallelismOne(localMember) uses
```

Comment:
can we do that in a different way? RootResultConsumerSink does many things:
- gathers all results
- implements `limit` and `offset`
- sends results to the client
 
Because of the last one it has to be executed on coordinator but other responsibilities would be better handled on member actually executing the query. Can we split that? There would have to be some network connection between "results collector" and "client sink" outside Jet job framework. Is that feasible? Otherwise, processor pruning might be used (@Fly-Style) - coordinator would run only `RootResultConsumerSink` and no other processor of the job.

Reply:
I propose to solve this issue it separate PR.